hyperspy/tests/io/test_hdf5.py

```diff
@@ -211,3 +212,19 @@ def test_rgba16():
 def test_strings_from_py2():
     s = EDS_TEM_Spectrum()
     nt.assert_equal(s.metadata.Sample.elements.dtype.char, "U")
+
+
+class TestPassingArgs:
+
+    def setUp(self):
+        self.filename = 'testfile.hdf5'
+        BaseSignal([1, 2, 3]).save(self.filename, compression_opts=8)
+
+    def test_compression_opts(self):
+        f = h5py.File(self.filename)
+        d = f['Experiments/__unnamed__/data']
+        nt.assert_equal(d.compression_opts, 8)
+        nt.assert_equal(d.compression, 'gzip')
```

Comment:
Upon exiting what?

From the h5py docs: "[the file] should be closed when they are no longer in use". I think that it'll be safer to close it.